client/my-sites/site-settings/site-icon-setting/index.jsx

```diff
@@ -104,9 +149,16 @@ class SiteIconSetting extends Component {
 									isVisible />
 							) }
 							siteId={ siteId }
-							onClose={ this.setSiteIcon }
+							onClose={ this.editSelectedMedia }
 							enabledFilters={ [ 'images' ] }
+							imageEditorProps={ {
+								allowedAspectRatios: [ AspectRatios.ASPECT_1X1 ],
+								onDone: this.setSiteIcon
+							} }
 							visible
+							labels={ {
+								confirm: translate( 'Continue' )
```

Comment:
>Should we add a context ? or are there any guidelines to know when we need to add a context or not

In this case, probably not. The WordPress Codex includes some good guidelines:

https://codex.wordpress.org/I18n_for_WordPress_Developers#Disambiguation_by_context

Typically my rules of thumb are:

- Use `context` if form is ambiguous, like "Edit" possibly being either a noun ("A post edit") and a verb ("Edit a post")
- Use `comment` if the string is otherwise confusing and to provide clarifying comments

My understanding is that it's good not to be overspecific with context and comments because they're treated as separate strings for translators.